controllers/machinehealthcheck_controller.go

```diff
@@ -274,7 +322,32 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 			t.string(),
 		)
 	}
+
 	for _, t := range healthy {
+		if m.Spec.RemediationTemplate != nil {
+			remediationObj := &corev1.ObjectReference{
+				APIVersion: m.Spec.RemediationTemplate.APIVersion,
+				Kind:       strings.TrimSuffix(m.Spec.RemediationTemplate.Kind, external.TemplateSuffix),
+				Name:       t.Machine.Name,
+			}
+
+			// Get remediation request object
+			obj, err := external.Get(ctx, r.Client, remediationObj, t.Machine.Namespace)
+			if err != nil {
+				if apierrors.IsNotFound(errors.Cause(err)) {
+					continue
+				}
+				return ctrl.Result{}, errors.Wrapf(err, "Failed to fetch remediation request")
```

Comment:
The first letter of an error message needs to be lowercase.

Should also probably include namespace, cluster name, machine name in the error message?

Do we surface this error in a condition anywhere? We need to make sure the log is not the sole place it goes.

Reply:
@ncdc how should we handle these errors in conditions?

Reply:
We need to decide where to put the information. It could probably go on the MHC, or Machine, or both. The goal is to make it easy for the user to understand there is a problem somewhere. @fabriziopandini what would you recommend?

Reply:
@ncdc 
If I got right this code we are cleaning up an external remediation request when the machine is back to healthy.
Doing this cleanup is an MHC responsibility, so this condition IMO should be on MHC.

However, If I look at when this error could happen and what to reports to the users, I'm not really sure this is should be a condition, because then the only reason I see `external.Get` failing is for temporary communication glitches, and so an error backoff should be fine to let the system to recover

Reply:
For this line in particular, you're right that it would be for a communications glitch, so maybe we don't need a condition here. My question is broader, though - for every place in this PR where we return an error, do any of them warrant conditions so the user is informed about the situation?

Reply:
Thanks for the clarification.  I have added a comment to the point were in my opinion there is an error that should be surfaced to the user